pkg/cmd/extension/manager.go

```diff
@@ -116,36 +117,96 @@ func (m *Manager) list(includeMetadata bool) ([]extensions.Extension, error) {
 		if !strings.HasPrefix(f.Name(), "gh-") {
 			continue
 		}
-		var remoteUrl string
-		updateAvailable := false
-		isLocal := false
-		exePath := filepath.Join(dir, f.Name(), f.Name())
-		if f.IsDir() {
-			if includeMetadata {
-				remoteUrl = m.getRemoteUrl(f.Name())
-				updateAvailable = m.checkUpdateAvailable(f.Name())
-			}
-		} else {
-			isLocal = true
-			if !isSymlink(f.Mode()) {
-				// if this is a regular file, its contents is the local directory of the extension
-				p, err := readPathFromFile(filepath.Join(dir, f.Name()))
-				if err != nil {
-					return nil, err
-				}
-				exePath = filepath.Join(p, f.Name())
-			}
+		ext, err := m.parseExtensionDir(f, includeMetadata)
+		if err != nil {
+			return nil, err
 		}
-		results = append(results, &Extension{
-			path:            exePath,
-			url:             remoteUrl,
-			isLocal:         isLocal,
-			updateAvailable: updateAvailable,
-		})
+		results = append(results, ext)
 	}
+
 	return results, nil
 }
 
+func (m *Manager) parseExtensionDir(fi fs.FileInfo, includeMetadata bool) (*Extension, error) {
+	id := m.installDir()
+	if _, err := os.Stat(filepath.Join(id, fi.Name(), manifestName)); err == nil {
+		return m.parseBinaryExtension(fi, includeMetadata)
+	}
+
+	return m.parseGitExtension(fi, includeMetadata)
+}
+
+func (m *Manager) parseBinaryExtension(fi fs.FileInfo, includeMetadata bool) (*Extension, error) {
```

Comment:
While slightly more verbose I think changing the name to `parseBinaryExtensionDir` is more clear, same goes for `parseGitExtension`.